tsuru/resource_tsuru_cluster.go

```diff
@@ -0,0 +1,147 @@
+package tsuru
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/tsuru/go-tsuruclient/pkg/tsuru"
+)
+
+func resourceTsuruCluster() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceTsuruClusterCreate,
+		ReadContext:   resourceTsuruClusterRead,
+		UpdateContext: resourceTsuruClusterUpdate,
+		DeleteContext: resourceTsuruClusterDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"addresses": {
+				Type: schema.TypeList,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Optional: true,
+			},
+			"tsuru_provisioner": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  "kubernetes",
+			},
+			"ca_cert": {
```

Comment:
Cluster all certificate related fields for easier readability.

Reply:
Maybe more verbose names, such as `ca_certificate`, `client_certificate`...

Reply:
@pedrokiefer I suggested this field to be compliant to tsuru types package.

Reply:
My take is that terraform should be easier to understand than the API. Keeping it equal to the types package you are exposing the API to another level of abstraction -- the provider should abstract this for our users.